### PR TITLE
#52 - flag as administrator

### LINKED ISSUE #52: Flag certain users as administrators
We need to restrict access to some parts of Brewmap to verified administrators. Nova has already implemented gate to do that: https://github.com/brewmap/brewmap/blob/develop/app/Providers/NovaServiceProvider.php#L39

My initial idea is to create a table `administrators` with user ids and timestamps. Probably it would be the cleanest solution, but of course I'm open to additional flags in `users` table or other ideas.

### PR DESCRIPTION
**Pull request by Baakoma**
My solution is the simplest possible. I just added a new field (is_admin) to the User model. I added an observer to ensure that the first created user is an admin.

NOTE:
If you want to test the functionality manually, you should change your env (APP_ENV) to something different, because nova skips checking gate('viewnova') on local environment.

It should close #52

### CODE REVIEW

File: features/http/dashboard.feature [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  Background:
+    Given there are users created:
+      | id                                   | email             | is_admin |
+      | 00000000-0000-0000-0000-000000000000 | admin@example.com | 1        |
+      | 00000000-0000-0000-0000-000000000001 | user@example.com  | 0        |
+
+  Scenario: User is requesting nova dashboard when he is not logged in
+    Given a user is requesting "/dashboard"
+    When a request is sent
+    Then a response status code should be "302"
```
Comment (krzysztofrewak) [inline features/http/dashboard.feature:13]: Why `302`?

File: app/Providers/NovaServiceProvider.php [hunk 2]
```diff
@@ -37,7 +37,7 @@ protected function gate(): void
         Gate::define(
             "viewNova",
             function (BrewmapUser $user): bool {
-                return true;
+                return $user->isAdmin;
```
Comment (krzysztofrewak) [inline app/Providers/NovaServiceProvider.php]: Probably we could shorten it into `Gate::define("viewNova", fn(BrewmapUser $user): bool => $user->isAdmin)`

**Review by krzysztofrewak — COMMENTED** [review]
I've found only two minor things. Nice work.

File: features/http/dashboard.feature (hunk 1 above)
Comment (mlencki) [reply features/http/dashboard.feature:13]: Isn't it default Nova behaviour? To throw `302 Redirect` if user is not logged in?

The same is in:
https://github.com/brewmap/brewmap/blob/97901c53a6a9d49e12526f554ad0e1b3f6ca301a/features/http/homepage.feature#L9-L12

Maybe then we should remove this scenario from `homepage.feature`?

Comment (krzysztofrewak) [reply features/http/dashboard.feature:13]: Ah, okay!

**Pull request merged**